defi/src/adaptors/db-utils/adaptor-record.ts

```diff
@@ -254,19 +269,20 @@ export const getAdaptorRecord = async (adaptorId: string, type: AdaptorRecordTyp
         expressionAttributeValues[":sk"] = timestamp
         keyConditionExpression = `${keyConditionExpression} and SK = :sk`
     }
-    try {
-        const resp = await dynamodb.query({
+    let resp: any
+    if (mode === "ALL") {
+        resp = await getHistoricalValues(adaptorRecord.pk)
```

Comment:
this works rn because all records are daily and thats what we query, but what about when we store these hourly? they'll just use a different id so wont be included here?

Reply:
my solution for this atm is, adding a flag to record `isIncomplete`, if it is true, that means this is not the final record for the day, otherwise use the same PK & SK as end of day 00:00. the flag can be used to track and fill if something breaks on 24th hour.

we can also store the record on `${PK}#hourly` with correct SK as backup, like what we are doing for tvl?